### PR TITLE
Mec172x ecia driver

### PR DESCRIPTION
**Pull request by scottwcpg**
A new driver for MEC172x aggregated interrupts. MEC172x and the older MEC152x implement an EC Interrupt Aggregator (ECIA) comprised of multiple GIRQ sub block that group interrupts. All GIRQs aggregate their interrupt inputs producing a single output to the ARM NVIC. A subset of GIRQs have a direct mode where the individual GIRQ result output (enable AND source) are connected to individual NVIC inputs. This driver handles GIRQs used in aggregated mode. The driver has a parent node for initializing the ECIA and GIRQ child nodes which are disabled by default. Enabling a GIRQ child node at build time causes the driver to allocate internal data structures for registering callbacks for each source implemented by the GIRQ. If an application does not need a GIRQ then it can be left disabled saving some memory. The PR is larger as this is a new driver requiring driver implementation, driver header files, DT bindings, DT chip description, and Kconfig settings.

### CODE REVIEW

**Comment by scottwcpg** [comment]
Rebased to address automated check failures.

File: drivers/interrupt_controller/intc_mchp_ecia_xec.c [hunk 1]
```diff
@@ -0,0 +1,1135 @@
...
+
+	return 0;
+}
+
+DEVICE_DT_DEFINE(DT_NODELABEL(girq8), xec_girq8_init,
+		 NULL, &xec_data_girq_8, &xec_config_girq_8,
+		 PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+		 NULL);
+#endif
+#if GIRQ09_AGGR
```
Comment (jackrosenthal) [inline drivers/interrupt_controller/intc_mchp_ecia_xec.c:496]: it seems this is severely repeated code

can these be consolidated to not do this?

File: drivers/interrupt_controller/intc_mchp_ecia_xec.c [hunk 2]
```diff
@@ -0,0 +1,1135 @@
...
+	const struct device *dev = get_girq_dev(girq_num);
+
+	return mchp_ecia_unset_callback_by_dev(dev, src);
+}
+
+#if (GIRQ08_AGGR || GIRQ09_AGGR || GIRQ10_AGGR || GIRQ11_AGGR ||\
+	GIRQ12_AGGR || GIRQ12_AGGR || GIRQ13_AGGR || GIRQ14_AGGR ||\
+	GIRQ15_AGGR || GIRQ16_AGGR || GIRQ17_AGGR || GIRQ18_AGGR ||\
+	GIRQ19_AGGR || GIRQ20_AGGR || GIRQ21_AGGR || GIRQ23_AGGR ||\
+	GIRQ24_AGGR || GIRQ25_AGGR || GIRQ26_AGGR)
```
Comment (jackrosenthal) [inline drivers/interrupt_controller/intc_mchp_ecia_xec.c:346]: this conditional is unwieldy and too large to maintain, please fix

**Review by jackrosenthal — CHANGES_REQUESTED** [review]

File: drivers/interrupt_controller/intc_mchp_ecia_xec.c (hunk 2 above)
Comment (scottwcpg) [reply drivers/interrupt_controller/intc_mchp_ecia_xec.c:346]: Please advise.  Do you want a another macro containing this macro?  The ISR is required any any of the nodes is enabled. No required if no node is enabled.  Without the conditional the compile complains of unused function when all the GIRQ nodes are disabled.

File: drivers/interrupt_controller/intc_mchp_ecia_xec.c (hunk 1 above)
Comment (scottwcpg) [reply drivers/interrupt_controller/intc_mchp_ecia_xec.c:496]: Maybe.  Each GIRQ had a different sized runtime data array.

Comment (scottwcpg) [reply drivers/interrupt_controller/intc_mchp_ecia_xec.c]: Found a way to consolidate with DT_FOREACH

File: drivers/interrupt_controller/intc_mchp_ecia_xec.c (hunk 2 above)
Comment (scottwcpg) [reply drivers/interrupt_controller/intc_mchp_ecia_xec.c]: Found a way to do this with DT_FOREACH removing all the GIRQxx_AGGR logic.

**Comment by scottwcpg** [comment]
Squashed into one commit.

**Review by dcpleung — APPROVED** [review]

**Review by jackrosenthal — APPROVED** [review]

**Pull request merged**